armi/reactor/components/component.py

```diff
@@ -1262,6 +1264,74 @@ def getIntegratedMgFlux(self, adjoint=False, gamma=False):
 
         return pinFluxes[self.p.pinNum - 1] * self.getVolume()
 
+    def getPinMgFluxes(
+        self, adjoint: Optional[bool] = False, gamma: Optional[bool] = False
+    ) -> np.ndarray:
+        """Retrieves the pin multigroup fluxes for the component.
+
+        Parameters
+        ----------
+        adjoint : bool, optional
+            Return adjoint flux instead of real
+        gamma : bool, optional
+            Whether to return the neutron flux or the gamma flux.
+
+        Returns
+        -------
+        np.ndarray
+            A ``(N, nGroup)`` array of pin multigroup fluxes, where ``N`` is the
+            equivalent to the multiplicity of the component (``self.p.mult``)
+            and ``nGroup`` is the number of energy groups of the flux.
+
+        Raises
+        ------
+        ValueError
+            If the location(s) of the component are not aligned with pin indices
+            from the block. This would happen if this component is not actually
+            a pin.
+        """
+        # Get the (i, j, k) location of all pins from the parent block
+        # FIXME: This should be changed to just using Block.getPinLocations once Drew's PR is merged
+        indicesAll = []
+        for clad in self.parent.getChildrenWithFlags(flags.Flags.CLAD):
+            if isinstance(clad.spatialLocator, grids.MultiIndexLocation):
+                indicesAll.extend(clad.spatialLocator.indices)
+            else:
+                indicesAll.append(clad.spatialLocator.indices)
+
+        # Retrieve the indices of this component
+        indices = self.spatialLocator.indices
+        if not isinstance(indices, list):
+            indices = [indices]
+
+        # Map this component's indices to block's pin indices
+        getIndex = lambda ind: next(
+            (i for i, indA in enumerate(indicesAll) if np.all(ind == indA)),
+            None,
+        )
+        indexMap = list(map(getIndex, indices))
+        # print(len(indices))
+        # print(len(indicesAll))
```

Comment:
Need to rm these old print statments. 

@zachmprince did you want to go through this and the test you wrote and do any other cleanup?